cm-ovirt-vm-creator.py

```diff
@@ -0,0 +1,193 @@
+import traceback
+import argparse
+import logging
+import time
+import ovirtsdk4 as sdk
+import ovirtsdk4.types as types
+
+
+
+
+def str2bool(s):
+    """ Convert str argument to bool """
+    if s.lower() in ('', 'yes', 'true', 't', 'y', '1'):
+        return True
+    elif s.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+
+
+logging.basicConfig(level=logging.DEBUG, filename="ocp-on-rhev-deployer.log")
+
+
+
+def doWork(args):
+    connection = None
+    try:
+        connection = sdk.Connection(url=args.ovirt_url , 
+                    username=args.ovirt_user, 
+                    password=os.environ[args.ovirt_pass], 
+                    ca_file=args.ovirt_ca_pem_file , 
+                    debug=True, 
+                    log=logging.getLogger())
+        system_service = connection.system_service()
+        vms_service = system_service.vms_service()
+        cluster_nodes = []
+        vm_name_template="{name_prefix}-{node_type}{i:03d}"
+        for idx in range(1, args.masters+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="master", i=idx))
+        for idx in range(1, args.infra_nodes+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="infra", i=idx))
+        for idx in range(1, args.nodes+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="compute", i=idx))
+        print (cluster_nodes)
+        if args.info:
+            getVmsInfo(connection, system_service, vms_service, cluster_nodes, args)
+        else:
+            createVMs(connection, system_service, vms_service, cluster_nodes, args)
+    except:
+        traceback.print_exc()
+    finally:
+        connection.close()
+
+
+
+
+def getVmsInfo(connection, system_service, vms_service, cluster_nodes, args):
+    vm_dict = {}
+    masters = []
+    infra_nodes=[]
+    nodes = []
+    for node in vmIterator(connection, system_service, vms_service, cluster_nodes):
+        node_name= node.get().name
+        print (node_name)
+        for dev in node.reported_devices_service().list():
+            if dev.name == 'eth0':
+                ips = dev.ips
+                for ip in ips:
+                    if ip.version == types.IpVersion.V4:
+                        vm_dict[node_name]=ip.address
+
+    if len(vm_dict) != len(cluster_nodes):
+        print (" PROBLEM - not all VMs were detected on the system ")
+        sys.exit(-1)
+    vm_names = vm_dict.keys()
+    vm_names.sort()
+    for vm_name in vm_names:
+        if vm_name.find("master") >= 0:
+            masters.append(vm_dict[vm_name])
+        elif vm_name.find("infra") >= 0:
+            infra_nodes.append(vm_dict[vm_name])
+        else:
+            nodes.append(vm_dict[vm_name])
+    print ("#################################################################")
```

Comment:
Also, the output format of `getVmsInfo()` and `createVMs()` is not consistent, we need to split this out into a function so that they'll both have the same output.